library/src/androidTest/java/org/xmtp/android/library/ConversationsTest.kt

```diff
@@ -208,4 +210,53 @@ class ConversationsTest {
             assertTrue(topics.contains(convo.topic))
         }
     }
+
+    @Test
+    fun testReturnsAllTopics() {
+        val key = SecureRandom().generateSeed(32)
+        val context = InstrumentationRegistry.getInstrumentation().targetContext
+        val eriWallet = PrivateKeyBuilder()
+
+        val eriClient = runBlocking {
+            Client().create(
+                account = eriWallet,
+                options = ClientOptions(
+                    ClientOptions.Api(XMTPEnvironment.LOCAL, false),
+                    appContext = context,
+                    dbEncryptionKey = key
+                )
+            )
+        }
+        val dm1 = runBlocking { eriClient.conversations.newConversation(boClient.address) }
+
+        val eriClient2 = runBlocking {
+            Client().create(
+                account = eriWallet,
+                options = ClientOptions(
+                    ClientOptions.Api(XMTPEnvironment.LOCAL, false),
+                    appContext = context,
+                    dbEncryptionKey = key,
+                    dbDirectory = context.filesDir.absolutePath.toString()
+                )
+            )
+        }
+        val dm2 = runBlocking { eriClient2.conversations.newConversation(boClient.address) }
+
+        runBlocking {
+            boClient.conversations.syncAllConversations()
+        }
+
+        val topics = boClient.conversations.allTopics()
+        val conversations = runBlocking { boClient.conversations.list() }
+        val hmacKeys = boClient.conversations.getHmacKeys()
+
+        assertEquals(topics.size, 2)
+        assertEquals(conversations.size, 1)
+        assertEquals(conversations.first().id, dm1.id)
```

Comment:
@codabrink what call needs to happen for eriClient1 and eriClient2 to both see the same dm instead of two different ones?

Reply:
Someone needs to send a message. That will cause the "primary DM" to be chosen.